rust/src/encodings/binary.rs

```diff
@@ -223,32 +232,38 @@ impl<'a, T: ByteArrayType> AsyncIndex<Range<usize>> for BinaryDecoder<'a, T> {
             .into_data()
         };
 
-        // Count nulls
-        let mut null_buf = MutableBuffer::new_null(self.length);
-        let mut null_count = 0;
-        int64_positions
-            .values()
-            .windows(2)
-            .enumerate()
-            .for_each(|(idx, w)| {
-                if w[0] == w[1] {
-                    bit_util::unset_bit(null_buf.as_mut(), idx);
-                    null_count += 1;
-                } else {
-                    bit_util::set_bit(null_buf.as_mut(), idx);
-                }
-            });
-
         let read_len = int64_positions.value(int64_positions.len() - 1) - start_position;
         let bytes = self
             .reader
             .get_range(start_position as usize..(start_position + read_len) as usize)
             .await?;
 
-        let array_data = ArrayDataBuilder::new(T::DATA_TYPE)
+        let mut data_builder = ArrayDataBuilder::new(T::DATA_TYPE)
             .len(index.len())
-            .null_count(null_count)
-            .null_bit_buffer(Some(null_buf.into()))
+            .null_count(0);
+
+        // Count nulls
+        if self.nullable {
+            let mut null_count = 0;
+            let mut null_buf = MutableBuffer::new_null(self.length);
+            int64_positions
```

Comment:
This feels like a big perf hit?

Reply:
This should still be way faster than reading a `bitmap` out of disk / s3.